eox_nelp/processors/xapi/event_transformers/initialized_events.py

```diff
@@ -22,26 +22,10 @@ class InitializedCourseTransformer(XApiTransformer):
         display=LanguageMap({constants.EN: constants.INITIALIZED}),
     )
 
-    def get_object(self):
+    @cached_property
+    def course_id(self):
         """
-        Get object for xAPI transformed event.
-
-        Returns:
-            `Activity`
+        Transformer's property that returns the associated
+        course id for the `nelc.eox_nelp.initialized.course` event.
         """
-        course_id = self.get_data('data.course_id', True)
-        object_id = self.get_object_iri('course', course_id)
-        course = get_course_from_id(course_id)
-        display_name = course["display_name"]
-        description = course["short_description"]
-        # Set default value if language is not found
-        course_language = course["language"] or eox_nelp_constants.DEFAULT_LANGUAGE
-
-        return Activity(
-            id=object_id,
-            definition=ActivityDefinition(
-                type=eox_nelp_constants.XAPI_ACTIVITY_COURSE,
-                name=LanguageMap(**({course_language: display_name} if display_name is not None else {})),
-                description=LanguageMap(**({course_language: description} if description is not None else {}))
-            ),
-        )
+        return self.get_data('data.course_id', True)
```

Comment:
Here you want to be required. Is difficulty to differentiate with the get of a dict so I thought it could be the default value.
https://github.com/nelc/event-routing-backends/blob/a3d132befcc70eb3548ff174adf018a421b8e690/event_routing_backends/processors/mixins/base_transformer.py#L155
If you want you can analyze this suggestion
```suggestion
        return self.get_data('data.course_id', required=True)
```